Add constructor for Troll object

scmx opened the pull request:
Lägger till en konstruktor för Troll.
För varje ny instans kan man välja varthän den ska hamna och vilket
maxHp den ska ha. Jag skapar en ny instans när ett troll dör och skickar
med vad det förra trollet hade för maxHp \* 2 när det nya skapas.
delay() metoden behöver lagas, se issue #13.
Skapade en variabel för positionen av trollSpawn.

scmx committed:
Add constructor for Troll object

Lägger till en konstruktor för Troll.
För varje ny instans kan man välja varthän den ska hamna och vilket
maxHp den ska ha. Jag skapar en ny instans när ett troll dör och skickar
med vad det förra trollet hade för maxHp * 2 när det nya skapas.
delay() metoden behöver lagas, se issue #13.
Skapade en variabel för positionen av trollSpawn.

alex-ross commented:
@CrazyDave1 var vaksam nu om du mergar både denna och min HealthBar samtidigt för då tror jag detta kommer faila.

scmx commented:
@CrazyDave1 HealthBar bör prioriteras först

scmx committed:
Rename wrongly named local variable troll -> enemy

Varibeln bör heta enemy eftersom det inte alltid är ett troll

scmx committed:
Add support for multiple trolls

Lägger till stöd för flera Troll. Allt som koden gjorde med
troll-variabeln har ändrats så att den loopar igenom arrayen med troll
istället och kör koden för varje troll.

scmx committed:
Increase number of trolls as score increases

Lägger till ett till troll när poängen går över 200, 500, 1000, 1700, ..
Det är ganska läskigt när trollet plötsligt delar sig itu och blir två
stycken.

scmx commented:
Lade till mer Troll-ändringar. Jag har skrivit en detaljerad beskrivning på varje commit, check em out.

David-Ahlander commented:
"We can’t automatically merge this pull request."
Add healthbar to troll if possible.

scmx commented:
@CrazyDave1 Jag fixade mergekonflikten

alex-ross commented:
@scmx tror även du får flytta health baren till din konstruktor.

David-Ahlander commented:
"We can’t automatically merge this pull request."

scmx committed:
Move healthBar init to troll constructor

scmx committed:
Fix bug where every troll shared the same hp

De delade på variabelb hp eftersom den låg i prototypen för Troll

scmx commented:
@CrazyDave1  Fixat

David-Ahlander commented:
Push it!

scmx commented:
@CrazyDave1 oops, fixat :)
